### PR TITLE
feat(tabs): deprecate TabsContainer component

### PR DESCRIPTION
**Pull request by austin94**
## Description

This PR deprecates the `TabsContainer` within the `react-tabs` package. It also migrates us to the `container-tabs` package.

## Detail

There should be no change in the API or consumption model for this component. There was some additional focus styling needed as we now use the "roving tab index" model of selection, but overall not too bad.

## Checklist

- [x] :ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)
- [x] :nail_care: view component styling is based on a Garden CSS
      component
- [x] :globe_with_meridians: Styleguidist demo is up-to-date (`yarn start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [x] :guardsman: includes new unit tests
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: packages/tabs/src/containers/TabsContainer.example.md [hunk 1]
```diff
@@ -1,82 +1,7 @@
...
-          {tab} content
-        </TabPanel>
-      ))}
-    </TabsView>
-  )}
-</TabsContainer>;
-```
+This component has been deprecated in favor of the API provided in the
+[@zendeskgarden/container-tabs](https://www.npmjs.com/package/@zendeskgarden/container-tabs)
+package.
```
Comment (ryanseddon) [inline packages/tabs/src/containers/TabsContainer.example.md]: Should you keep the examples here until it's removed?

File: packages/tabs/src/views/Tab.js [hunk 2]
```diff
@@ -28,6 +28,16 @@ const Tab = styled.div.attrs(props => ({
     [TabStyles['is-selected']]: props.selected
   })
 }))`
+  &&& {
+    ${props =>
+      !props.focused &&
+      `
+      :before {
+        box-shadow: none;
```
Comment (ryanseddon) [inline packages/tabs/src/views/Tab.js]: Is this fixing a bug? Why is this included?

File: packages/tabs/src/elements/Tabs.js [hunk 3]
```diff
@@ -8,16 +8,17 @@
...
-import TabsContainer from '../containers/TabsContainer';
 import TabsView from '../views/TabsView';
 import TabList from '../views/TabList';
 import Tab from '../views/Tab';
 
 /**
  * High-level abstraction for basic Tabs implementations.
  */
-export default class Tabs extends ControlledComponent {
+class Tabs extends ControlledComponent {
```
Comment (ryanseddon) [inline packages/tabs/src/elements/Tabs.js]: Why not convert to a functional component here and use `useTabs` instead? It helps reduce code size and is easier to read. See #431

File: packages/tabs/src/containers/TabsContainer.example.md (hunk 1 above)
Comment (austin94) [reply packages/tabs/src/containers/TabsContainer.example.md]: Sure, I will add that back.

File: packages/tabs/src/elements/Tabs.js (hunk 3 above)
Comment (austin94) [reply packages/tabs/src/elements/Tabs.js]: This would be a breaking change with removing `onStateChange` and how people could have overrode it.

We can include that API change when we change the Tabs API in CSS-in-JS.

**Comment by austin94** [comment]
> People will have a lot of warnings in tests and dev if we push this into a minor release

I originally had the warning at the root of the file, which would have shown the warning at the root import. I've now moved this to within the `TabsContainer` component (and the previous deprecation in `BreadcrumbsContainer`) to reduce the visibility to those that only use it directly.

File: packages/tabs/src/views/Tab.js (hunk 2 above)
Comment (jzempel) [reply packages/tabs/src/views/Tab.js]: Feels like this hack should be commented in code.

Comment (austin94) [reply packages/tabs/src/views/Tab.js]: I've added a comment for this.
